packages/node-webpack-startup-snapshot-checker/src/index.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable no-console */
+import { execFile as execFileCallback } from 'child_process';
+import { promisify } from 'util';
+import os from 'os';
+import path from 'path';
+import { promises as fs } from 'fs';
+import assert from 'assert';
+import { createRequire } from 'module';
+const execFile = promisify(execFileCallback);
+
+(async function main() {
+  const target = process.argv[2];
+  if (!target) {
+    throw new Error(
+      'Usage: npx node-webpack-startup-snapshot-checker <filename>'
+    );
+  }
+
+  const tmpdir = path.join(
+    os.tmpdir(),
+    `node-webpack-startup-snapshot-checker-${Date.now()}`
+  );
+  try {
+    const entryFile = path.join(tmpdir, 'entry.js');
+    const webpackConfig = path.join(tmpdir, 'webpack.config.js');
+    await fs.mkdir(tmpdir, { recursive: true });
+
+    await fs.writeFile(
+      entryFile,
+      `
+'use strict';
+const v8 = require('v8');
+const target = require(${JSON.stringify(
+        createRequire(process.cwd() + '/dummy.js').resolve(target)
+      )});
+
+function printTarget() {
+  console.log(typeof target);
+  if ((typeof target === 'object' || typeof target === 'function') && target) {
+    for (const [key, value] of Object.entries(target)) {
+      console.log(\`\${key} \${typeof value}\`);
+    }
+  }
+}
+
+if (v8.startupSnapshot.isBuildingSnapshot()) {
+  v8.startupSnapshot.setDeserializeMainFunction(() => printTarget());
+} else {
+  printTarget();
+}
+`
+    );
+
+    await fs.writeFile(
+      webpackConfig,
+      `
+'use strict';
+const path = require("path");
+module.exports = {
+  devtool: false,
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'webpacked.js',
+    library: { name: 'entry', type: 'var' }
+  },
+  module: {
+    rules: [
+      {
+        test: /\\.node$/,
+        loader: "null-loader",
```

Comment:
Do we want to silence all node modules? Or what exactly here is happening?

Reply:
Yup, exactly that 👍 They can't be snapshotted anyway